servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ConnectionObserverEventOrderTest.java

```diff
@@ -0,0 +1,207 @@
+                    .appendClientFilter(new HttpLifecycleObserverRequesterFilter(new HttpLifecycleObserverImpl()))
+                    .appendClientFilter(new RetryingHttpRequesterFilter.Builder()
+                            .retryRetryableExceptions((req, ex) ->
+                                    RetryingHttpRequesterFilter.BackOffPolicy.ofNoRetries())
+                            .build())
+                    .buildBlocking()) {
+                serverContext.close(); // causes connection establishment to fail.
+                HttpResponse response = client.request(client.get("/sayHello"));
+                response.payloadBody();
+            }
+        });
+
+        List<String> expectedEvents = Arrays.asList("onNewExchange", "onRequest", "onNewConnection",
+                "connectionClosed", "onResponseError", "onExchangeFinally");
+        for (String expected : expectedEvents) {
+            assertEquals(expected, eventQueue.take());
+        }
+    }
+
+    private final class HttpLifecycleObserverImpl implements HttpLifecycleObserver {
+        @Override
+        public HttpExchangeObserver onNewExchange() {
+            addEvent();
+            return new HttpExchangeObserver() {
+                @Override
+                public void onConnectionSelected(ConnectionInfo info) {
+                    addEvent();
+                }
+
+                @Override
+                public HttpRequestObserver onRequest(HttpRequestMetaData requestMetaData) {
+                    addEvent();
+                    return NoopHttpLifecycleObserver.NoopHttpRequestObserver.INSTANCE;
+                }
+
+                @Override
+                public HttpResponseObserver onResponse(HttpResponseMetaData responseMetaData) {
+                    addEvent();
+                    return NoopHttpLifecycleObserver.NoopHttpResponseObserver.INSTANCE;
+                }
+
+                @Override
+                public void onResponseError(Throwable cause) {
+                    addEvent();
+                }
+
+                @Override
+                public void onResponseCancel() {
+                    addEvent();
+                }
+
+                @Override
+                public void onExchangeFinally() {
+                    addEvent();
+                }
+            };
+        }
+    }
+
+    private final class TransportObserverInjectorFilter<R>
+            implements ConnectionFactoryFilter<R, FilterableStreamingHttpConnection> {
+
+        @Override
+        public ConnectionFactory<R, FilterableStreamingHttpConnection> create(
+                ConnectionFactory<R, FilterableStreamingHttpConnection> original) {
+            return new TransportObserverInjector<>(original);
+        }
+
+        @Override
+        public ExecutionStrategy requiredOffloads() {
+            return ExecutionStrategy.offloadNone();
+        }
+    }
+
+    private final class TransportObserverInjector<R> extends
```

Comment:
Seems quite similar to `TransportObserverConnectionFactoryFilter`. Can we use the existing one?